akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -250,6 +250,11 @@ object Source {
    * The actor will be stopped when the stream is completed, failed or canceled from downstream,
    * i.e. you can watch it to get notified when that happens.
    *
+   * An async boundary is automatically created after the returned actorRef source, so downstream
+   * operations will not occur in the same thread as the Actor ingesting messages. If you select too
```

Comment:
I think this comment is wrong "so downstream operations will not occur in the same thread as the Actor ingesting messages".

The async boundary between the sender and the `ActorRefSourceActor` is the actor and its mailbox itself, and that is what detaches the sender thread from the stream processing. The async boundary after the stage has nothing to do with sender's thread.